test/test_v2_dataset_warnings.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+
+
+def test_warns_if_imported_from_datasets():
+    import torchvision
+
+    value = torchvision._WARN_ABOUT_BETA_TRANSFORMS
+
+    setattr(torchvision, "_WARN_ABOUT_BETA_TRANSFORMS", True)
+
+    with pytest.warns(UserWarning, match=torchvision._BETA_TRANSFORMS_WARNING):
+        from torchvision.datasets import wrap_dataset_for_transforms_v2
+
+    setattr(torchvision, "_WARN_ABOUT_BETA_TRANSFORMS", value)
+
+
+def test_no_warns_if_imported_from_datasets():
```

Comment:
This test is failing, but I'd suggest removing both tests.
They can interfer with the import and warnings in ways that we may not predict properly, and this may lead to surprising or undesired behaviour (like masking a warning it shouldn't mask)